### PR TITLE
chore(STONEO11Y-104): Add shellcheck to the CI pipeline

### PR DESCRIPTION
**Pull request by Kousalya1998**
* Added shellcheck lint to Makefile.
* Added tekton task named shellscript-lint

### CODE REVIEW

**Review by hmariset — APPROVED** [review]
LGTM

**Comment by eisraeli** [comment]
Hi @Kousalya1998 

1. Notice you have 4 commits on the PR (while there should have been only 1).

2. Please add a short description of what is being added, in this case, ShellCheck.
For example:
"ShellCheck is an open source static analysis tool that automatically finds bugs in your shell scripts."

File: Makefile [hunk 1]
```diff
@@ -69,3 +69,12 @@ kustomize_build:
 	# It will fail once we have more than one subdirectory, which will prevent us from
 	# adding untested configurations (this target will have to chage when that happens).
 	./kustomize build prometheus/* 1>/dev/null
+
+.PHONY: install_shellcheck
+install_shellcheck:
+	yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
```
Comment (hmariset) [inline Makefile:75]: can you replace yum with dnf (just a suggestion)

File: Makefile [hunk 2]
```diff
@@ -69,3 +69,12 @@ kustomize_build:
...
+	yum install -y ShellCheck
```
Comment (hmariset) [inline Makefile:76]: same as above

File: Makefile [hunk 3]
```diff
@@ -69,3 +69,12 @@ kustomize_build:
...
+
+.PHONY: lint_shellscript
+lint_shellscript:
+	find . -type f -name "*.sh" -exec shellcheck {} + && echo "lint_shellcheck: SUCCESS"
```
Comment (srivickynesh) [inline Makefile:80]: Where is the test it's running against?

Comment (Kousalya1998) [reply Makefile:80]: Its linting all shell scripts in a directory and its subdirectories using this command @srivickynesh

Comment (srivickynesh) [reply Makefile:80]: There are no shell scripts all are converted to makefile.

File: Makefile (hunk 1 above)
Comment (yftacherzog) [reply Makefile:75]: We've been using yum for this repo. I don't think we should have yum in one place and dnf in another. If we prefer to use dnf, then I think we should have everything changed to dnf across the repo at once.

**Review by yftacherzog — CHANGES_REQUESTED** [review]
Few points:

As @srivickynesh mentioned, this tool currently have no effect as we're not using any shell scripts within the repo, so by introducing it, we're introducing dead code to our CI, trying to foresee the future. @amisstea i see you created the ticket, can you elaborate on the rationale for doing this at this time?

Regarding the implementation, the way this PR is implemented at the moment, we're following 2 different approaches for developers and for CI: we ask developers to install it on their own (which is another hurdle for onboarding we're introducing for something that is not even used for anything), while on CI we install it automatically.

I don't think we should have it installed for developers system-wide using yum, but I think there's an alternative:

Looking at the installation guide: https://github.com/koalaman/shellcheck#installing

I see the section starts with: "Alternatively, you can download pre-compiled binaries for the latest release here".

I think this will be a more suited approach for our case, as it will allow us to:
1. Download a binary which can be run directly instead of installing it system-wide.
2. Separate the step of installing the tool from the step of using it (as it will be a binary and not part of the system config).
3. Consolidate CI and local setup in a similar way to previous tools we included.
4. Reduce the additional dependency for dev envs that is being added to the contributing guide.

Other than that, following what @eisraeli said, I think this PR is missing context and motivation for _why_ are we making those changes. It will allow the reviewers to understand what is the tool we're adding and why we're adding it. Being able to answer this before creating the PR, have the potential to also eliminate some of the questions raised here, and in turn eliminate some review cycles.

**Comment by Kousalya1998** [comment]
@amisstea Alex, I need your input here.

**Comment by yftacherzog** [comment]
> @amisstea Alex, I need your input here.

Alex mentioned on the stand up call that he has no additional information about why this is needed.

**Comment by amisstea** [comment]
> @amisstea Alex, I need your input here.

We created this story in a refinement session soon after https://issues.redhat.com/browse/STONEO11Y-90 was complete. We probably failed to recognize this at the time. IMO we shouldn't add functionality to the CI which isn't utilized because then it's only tech debt providing no value. I'm all in favor of closing this one.

**Comment by Kousalya1998** [comment]
I am closing this pull request at the moment.

**Pull request closed without merging**